flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/writer/ResultPartitionWriter.java

```diff
@@ -63,7 +65,16 @@
 	 *
 	 * @return true if operation succeeded and bufferConsumer was enqueued for consumption.
 	 */
-	boolean addBufferConsumer(BufferConsumer bufferConsumer, int subpartitionIndex) throws IOException;
+	boolean addBufferConsumer(BufferConsumer bufferConsumer, int subpartitionIndex, boolean insertInFront) throws IOException;
```

Comment:
`insertInFront` is actually `insertAtEnd`, right?

Reply:
I also thought `insertInFront` bad while implementing. But for PoC I did not considered much at that time. `insertAsHead` is indeed better and i will take it.